src/components/Main/CustomDialog/CustomDialog.js

```diff
@@ -0,0 +1,152 @@
+import React, { PureComponent } from 'react'
+import { withTranslation } from 'react-i18next'
+import moment from 'moment-timezone'
+import {
+  Button,
+  Classes,
+  Dialog,
+  Intent,
+  Position,
+} from '@blueprintjs/core'
+import { DateRangeInput } from '@blueprintjs/datetime'
+
+import { DEFAULT_DATETIME_FORMAT, momentFormatter } from 'state/utils'
+
+import { propTypes, defaultProps } from './CustomDialog.props'
+
+const SMALL_DATE_RANGE_POPOVER_PROPS = {
+  position: Position.TOP,
+}
+
+class CustomDialog extends PureComponent {
+  componentDidMount() {
+    this.maxDate = new Date()
```

Comment:
any reason to set maxDate in DidMount ? 
I feel that this should be in constructor otherwise we will get `undefined` before mount.

Reply:
doesn't seem to be a required prop and gets set on second render when modal is opened
changed to recalculation on each render

Reply:
okay :D